### PR TITLE
Array lab

### PR DESCRIPTION
**Pull request by sevfitz**

### CODE REVIEW

File: src/array-methods.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    for (let i = 0; i < array.length; i++) {
+        accumulator = callback(accumulator, array[i], i);
+    }
+    return accumulator;
+}
+
+function findIndexMethod(array, callback) {
+    for (let i = 0; i < array.length; i++) {
+        if (callback(array[i], i) == true) return i;
```
Comment (martypdx) [inline src/array-methods.js]: Just use:  `if (callback(array[i], i)) return i;`

File: tests/array-methods.tests.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        assert.deepEqual(results, [1, 2, 5]);
+    });
+
+    it('map: it takes a callback and applies it to an array, then returns the new array', () => {
+        const testArray = [1,2,5];
+        const results = [];
+        arrayMethods.mapMethod(testArray, (item, i) => { 
+            results[i] = item + 1;
+        });
+        assert.deepEqual(results, [2, 3, 6]);
```
Comment (martypdx) [inline tests/array-methods.tests.js]: You should be testing the return value from `map`

File: tests/array-methods.tests.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        const testArray = ['hello', 'goodbye', 'surprise'];
+        let index = 0;
+        arrayMethods.findIndexMethod(testArray, (item, i) => {
+            if (item == 'goodbye') {
+                index = i;
+                return true;
+            }
+        });
+
+        assert.deepEqual(index, 1);
```
Comment (martypdx) [inline tests/array-methods.tests.js]: test return value

File: tests/array-methods.tests.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    });
+
+    it('every: it takes a callback and returns true if each item of the array returns a truthy value after the callback is applied', () => {
+        const testArray = [3,2, 0, 5,1];
+        let everyTest = arrayMethods.everyMethod(testArray, (item, i) => {
+            if (item < 5) return true;
```
Comment (martypdx) [inline tests/array-methods.tests.js]: Just: `return item < 5;`

File: tests/array-methods.tests.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                index = i;
+                return true;
+            }
+        });
+
+        assert.deepEqual(index, 1);
+    });
+
+    it('every: it takes a callback and returns true if each item of the array returns a truthy value after the callback is applied', () => {
+        const testArray = [3,2, 0, 5,1];
```
Comment (martypdx) [inline tests/array-methods.tests.js]: What's up with that spacing inside the array?

**Comment by nsmeds** [comment]
You're attempting to merge into the organization's `master` branch, rather than the `sevfitz` branch - can you update this? Let me know if you need help.

**Pull request merged**